src/platforms/Poap/PoapCardButton.tsx

```diff
@@ -0,0 +1,70 @@
+import { ButtonProps, Tooltip } from "@chakra-ui/react"
+import { getTimeDiff } from "components/[guild]/RolePlatforms/components/PlatformCard/components/AvailabilityTags"
+import useGuild from "components/[guild]/hooks/useGuild"
+import useGuildPermission from "components/[guild]/hooks/useGuildPermission"
+import Button from "components/common/Button"
+import LinkButton from "components/common/LinkButton"
+import platforms from "platforms/platforms"
+import { GuildPlatform } from "types"
+
+type Props = {
+  platform: GuildPlatform
+}
+
+const PoapCardButton = ({ platform }: Props) => {
+  const { urlName, roles } = useGuild()
+  const { isAdmin } = useGuildPermission()
+
+  const rolePlatform = roles
+    ?.find((r) => r.rolePlatforms.some((rp) => rp.guildPlatformId === platform.id))
+    ?.rolePlatforms?.find((rp) => rp.guildPlatformId === platform?.id)
+
+  const startTimeDiff = getTimeDiff(rolePlatform?.startTime)
+  const endTimeDiff = getTimeDiff(rolePlatform?.endTime)
+
+  const isButtonDisabled =
```

Comment:
This logic is repeated in multiple places (NFT, text rewards, POAP), but didn't want to add more changes to this PR, so I made a Linear issue, will make sure to make a hook for this. 👍